File: TaskFlow.Web/src/lib/hubEvents.ts
// Mirror of the backend's TaskFlow.Api/Hubs/HubEvents.cs. These strings are a cross-language
// contract with the C# hub: a typo on either side silently stops the live feed. There is no
// shared type across the language boundary, so each side keeps its own copy.
export const HubEvents = {
  AgentAction: 'AgentAction',
  AgentCycle: 'AgentCycle',
} as const

File: TaskFlow.Api/Hubs/HubEvents.cs
namespace TaskFlow.Api.Hubs;

/// <summary>
/// SignalR event names. These strings are a contract with the React client
/// (see useAgentFeed's connection.on handlers) — a typo on either side silently
/// stops the live feed, so keep the two ends in sync.
/// </summary>
public static class HubEvents
{
    public const string AgentAction = "AgentAction";
    public const string AgentCycle = "AgentCycle";
}
